compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -745,7 +745,15 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         if object_safety_violations {
             return;
         }
+
+        // related to issue #91997, turbofishes added only when in an expr or pat
+        let mut in_expr_or_pat = false;
         if let ([], [bound]) = (&potential_assoc_types[..], &trait_bounds) {
+            let grandparent = tcx.parent_hir_node(tcx.parent_hir_id(bound.trait_ref.hir_ref_id));
+            in_expr_or_pat = match grandparent {
+                Node::Expr(_) | Node::Pat(_) => true,
+                _ => false,
+            };
```

Comment:
We still suggest a turbofish for `<MyIterator>::next` where it's not strictly necessary (contrary to `MyIterator::next`) but I don't think it matters. We *could* check if the parent of the trait_ref is a `Path(TypeRelative(_, segment))` where `segment.infer_args` is `false` but er, that's a bit hacky.